include/z64bgcheck.h

```diff
@@ -52,10 +52,51 @@ typedef struct {
 } CollisionPoly; // size = 0x10
 
 typedef struct {
-    /* 0x00 */ u16 cameraSType;
-    /* 0x02 */ s16 numCameras;
-    /* 0x04 */ Vec3s* camPosData;
-} CamData;
+    /* 0x00 */ Vec3s pos;
+    /* 0x06 */ Vec3s rot;
+    /* 0x0C */ s16 fov;
+    /* 0x0E */ union {
+                s16 jfifId;
+                s16 timer;
+                s16 flags;
+    };
+    /* 0x10 */ s16 unk_10; // unused
+} SubBgCamData;
+
+/**
+ * BgCamData Summary:
+ * 
+ * setting - camera setting described by CameraSettingType enum
+ * numData - The total count of Vec3s data in the collision
+ * data - data stored in Vec3s with various purposes summarized below:
+ * 
+ * numData = 0:
+ *      data unused
+ * 
+ * numData = 3: data organized in SubBgCamData struct
+ *      data[0]   // Position
+ *      data[1]   // Rotation
+ *      data[2].x // Field of View
+ *      data[2].y // Jfif Id, timer, flags
+ *      data[2].z // unused
+ * 
+ * numData = 6: Crawlspaces only (CAM_SET_CRAWLSPACE), entirely position data
+ *      data[1] // Front entrance coordinates to crawlspace
+ *      data[4] // Back entrance coordinates to crawlspace
+ *      data[0], data[2], data[3], data[5] // Unused coordinates along crawlspace line
```

Comment:
For this section and the numData = 9 one, I suggest simply describing the data as an array of points along the crawlspace, and specifically mentioning indices 1 and 4 (or 7) as the entrances. I think this would keep it more concise and make it clearer that the points are all in order along the line and could technically be used.

Also I don't see anything in the code that limits these to numData 6 or 9 specifically, so it might even make sense to just explain that the data represents points along the crawlspace when used with `CAM_SET_CRAWLSPACE`, in which case indices 1 and numData - 2 are used as the entrances (based on `Camera_Subj4`).
My reasoning is that in theory this data could be used with any amount of points you want during modding, and the other points could be used to implement a more advanced crawlspace feature.
This last part might not be worth the trouble tho, just an idea.

Reply:
Yeah, as a follow-up PR, I was going to be documenting crawlspaces with a section above. I could just have the description in one place, above `Camera_Subj4`, and use your suggestions to make this explanation more concise (I might add that description to this PR to work out the flow?)
https://github.com/engineer124/oot/blob/79331ed86537be0ca09157854617ae17bf658929/src/code/z_camera.c#L4303-L4323